Move TextIndicatorWindow to WebKitLegacy.

megangardner opened the pull request:
#### 5051d18164d2e404dcf6d5e35801616a6bd8edc8
<pre>
Move TextIndicatorWindow to WebKitLegacy.
<a href="https://bugs.webkit.org/show_bug.cgi?id=293699">https://bugs.webkit.org/show_bug.cgi?id=293699</a>
<a href="https://rdar.apple.com/152182885">rdar://152182885</a>

Reviewed by Wenson Hsieh.

Now that TextIndicatorWindow is only used in WebKitLegacy
it should live there instead of WebCore.

* Source/JavaScriptCore/DerivedSources-input.xcfilelist:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/TextIndicator.h:
* Source/WebCore/page/scrolling/mac/ScrollerMac.h:
* Source/WebKitLegacy/SourcesCocoa.txt:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/WebCoreSupport/TextIndicatorWindow.h: Renamed from Source/WebCore/page/mac/TextIndicatorWindow.h.
* Source/WebKitLegacy/mac/WebCoreSupport/TextIndicatorWindow.mm: Renamed from Source/WebCore/page/mac/TextIndicatorWindow.mm.
(-[WebTextIndicatorView isFlipped]):
(TextIndicatorWindow::TextIndicatorWindow):
(TextIndicatorWindow::~TextIndicatorWindow):
(TextIndicatorWindow::setAnimationProgress):
(TextIndicatorWindow::clearTextIndicator):
(TextIndicatorWindow::setTextIndicator):
(TextIndicatorWindow::updateTextIndicator):
(TextIndicatorWindow::closeWindow):
(TextIndicatorWindow::startFadeOut):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _setTextIndicator:withLifetime:]):
* Source/WebKitLegacy/mac/WebView/WebViewData.h:
* Source/WebKitLegacy/mac/WebView/WebViewData.mm:
* Source/WebKitLegacy/mac/WebView/WebViewInternal.h:

Canonical link: <a href="https://commits.webkit.org/295522@main">https://commits.webkit.org/295522@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d0c76ec81101e8cd6fa930aea75e24bf8eaa8032

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105342 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25054 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/15481 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/110550 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/55999 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/107383 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/25492 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/33597 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80010 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/55999 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108348 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/19883 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/95077 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60318 "run-api-tests-without-change (failure)") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13155 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/55394 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/89337 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13197 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/113209 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/32497 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/23953 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89086 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/32860 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91295 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88727 "Found 100 new API test failures: /TestWebKit:WebKit.CanHandleRequest, /WebKitGTK/TestGeolocationManager:/webkit/WebKitGeolocationManager/watch-position, /TestWebKit:WebKit.UserMediaBasic, /TestWebKit:WebKit.PageLoadEmptyURL, /WebKitGTK/TestOptionMenu:/webkit/WebKitWebView/option-menu-groups, /TestWebKit:WebKit.GetInjectedBundleInitializationUserDataCallback, /TestWebKit:WebKit.GeolocationTransitionToHighAccuracy, /TestWebKit:WebKit.PendingAPIRequestURL, /TestWebKit:WebKit.FirstMeaningfulPaint, /WebKitGTK/TestUIClient:/webkit/WebKitWebView/file-chooser-request ... (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22628 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/33619 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11409 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/27972 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/32421 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/37835 "Found 1 new failure in mac/WebCoreSupport/TextIndicatorWindow.mm and found 1 fixed file: page/mac/TextIndicatorWindow.mm") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/32193 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/35538 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/33767 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKitLegacy/SourcesCocoa.txt:
```diff
@@ -168,6 +168,7 @@ mac/Plugins/WebPluginDatabase.mm
 
 mac/History/WebBackForwardList.mm
 mac/WebCoreSupport/WebFrameLoaderClient.mm
+mac/WebCoreSupport/TextIndicatorWindow.mm
```
Nit - alphabetical order

megangardner committed:
Move TextIndicatorWindow to WebKitLegacy.
https://bugs.webkit.org/show_bug.cgi?id=293699
rdar://152182885

Reviewed by Wenson Hsieh.

Now that TextIndicatorWindow is only used in WebKitLegacy
it should live there instead of WebCore.

* Source/JavaScriptCore/DerivedSources-input.xcfilelist:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/TextIndicator.h:
* Source/WebCore/page/scrolling/mac/ScrollerMac.h:
* Source/WebKitLegacy/SourcesCocoa.txt:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/WebCoreSupport/TextIndicatorWindow.h: Renamed from Source/WebCore/page/mac/TextIndicatorWindow.h.
* Source/WebKitLegacy/mac/WebCoreSupport/TextIndicatorWindow.mm: Renamed from Source/WebCore/page/mac/TextIndicatorWindow.mm.
(-[WebTextIndicatorView isFlipped]):
(TextIndicatorWindow::TextIndicatorWindow):
(TextIndicatorWindow::~TextIndicatorWindow):
(TextIndicatorWindow::setAnimationProgress):
(TextIndicatorWindow::clearTextIndicator):
(TextIndicatorWindow::setTextIndicator):
(TextIndicatorWindow::updateTextIndicator):
(TextIndicatorWindow::closeWindow):
(TextIndicatorWindow::startFadeOut):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _setTextIndicator:withLifetime:]):
* Source/WebKitLegacy/mac/WebView/WebViewData.h:
* Source/WebKitLegacy/mac/WebView/WebViewData.mm:
* Source/WebKitLegacy/mac/WebView/WebViewInternal.h:

Canonical link: https://commits.webkit.org/295522@main
